kernel/sched.c

```diff
@@ -827,6 +827,7 @@ int z_pend_curr(struct k_spinlock *lock, k_spinlock_key_t key,
 	 * held.
 	 */
 	(void) k_spin_lock(&sched_spinlock);
+	_current->obj_lock = lock;
```

Comment:
Is this not still racy?  You hold the scheduler lock here, but the unpend/timeout code that is going to end up reading this value does not.